src/router/middlewares/setTenantId.ts

```diff
@@ -11,6 +11,20 @@ import RouteHelper from '../routes/routeHelper';
 
 const tenantIdRegex = /^[a-zA-Z0-9\-_]{1,64}$/;
 
+const getTenantIdFromAudClaim = (audClaim: string, baseUrl: string) => {
+    // aud claim should ALWAYS be present
+    if (!audClaim) {
+        throw new UnauthorizedError('Unauthorized');
+    }
+    if (audClaim === baseUrl) {
+        return undefined;
+    }
+    if (audClaim.startsWith(`${baseUrl}/tenant/`)) {
+        return audClaim.substring(`${baseUrl}/tenant/`.length);
+    }
+    throw new UnauthorizedError('Unauthorized');
+};
```

Comment:
I think that this method can be changed to something like:
```js
const getTenantIdFromAudClaim = (audClaim: string, baseUrl: string) => { 
    if (!audClaim) {
        return undefined;
    }
    if (audClaim.startsWith(`${baseUrl}/tenant/`)) {
        return audClaim.substring(`${baseUrl}/tenant/`.length);
    }
    return undefined;
};
```
In a sense it is a best effort to extract the tenantId from the `aud` claim.

Checking the existence and the validity of the `aud` claim is done on the [SMART package](https://github.com/awslabs/fhir-works-on-aws-authz-smart/blob/c644cb7a867d456cebb587a1f3cf6d036b6226af/src/smartAuthorizationHelper.ts#L140) or by the APIGW + Cognito intergration, so there's no need to redo it here.

Also, the cognito token doesn't use urls for `aud`. Cognito will always use custom claims for the tenantId and the router should not fail if the `aud` claim doesn't follow the `<baseUrl>/tenant/<tenantId>` schema.

```js
{
  "sub": "5ecb250e-8964-4120-ae25-xxxxxxxxx,
  "cognito:groups": [
    "practitioner"
  ],
  "email_verified": true,
  "iss": "https://cognito-idp.us-west-2.amazonaws.com/us-west-2_xxxxxxxx",
  "cognito:username": "workshopxxxxx",
  "origin_jti": "7467505b-e196-431d-b069-xxxxxxxxxx",
  "custom:tenantId": "tenant1",
  "aud": "2qi5m0hlnp4sxxxxxxxxxxxx",
  "event_id": "54f405a6-7e15-415e-b4f7-xxxxxxxxxxx",
  "token_use": "id",
  "auth_time": 1625138033,
  "exp": 1625141633,
  "iat": 1625138033,
  "jti": "07ef9e6a-b823-401b-bae1-xxxxxxxxxxx",
  "email": "dummy@email.com"
}
```

Sorry if all of this was not clear on the design doc.

Reply:
Also, the `aud` claim can be a string or an array of strings.

I'm not sure if our Okta implementation ever uses an array of strings for `aud`, but we should handle both cases for compatibility with other IDP setups.